### PR TITLE
Dismemberment from explosion

### PR DESCRIPTION
**Pull request by Badulundabad**
- added CannonBallScript runs on collision, causes damage depending on distance from explosion to victim and runs visual and sound effects. This script tears enemy apart if it's killed by the explosion and closer than 3 meters (or whatever);
- added SmokingLimbScript that runs smoke on body parts for some short time;
- added smoke particles to cannon shot;
- added calculation of blow direction in Agent.ApplyDamage extension method. It helps to knock down enemy in right direction;
- added exploded_body_parts xml with implemented body parts are created by @RagnarHrodgarson;
- added static method EnableSlowMotion to TestingDamageMissionLogic.

### CODE REVIEW

File: CSharpSourceCode/Utilities/Extensions/AgentExtensions.cs [hunk 1]
```diff
@@ -380,7 +381,13 @@ public static void ApplyDamage(this Agent agent, int damageAmount, Agent damager
                     if (damager != null)
                     {
                         var checkAgent = Mission.Current.FindAgentWithIndex(damager.Index);
-                        if (checkAgent != null && checkAgent.Equals(damager)) blow.OwnerId = damager.Index;
+                        if (checkAgent != null && checkAgent.Equals(damager))
+                        {
+                            blow.OwnerId = damager.Index;
+                            blow.Position = impactPosition;
+                            blow.Direction = agent.Position - impactPosition;
```
Comment (hunharibo) [inline CSharpSourceCode/Utilities/Extensions/AgentExtensions.cs:388]: you should probably normalize this.

**Review by hunharibo — APPROVED** [review]
good job, one minor comment.

File: CSharpSourceCode/Utilities/Extensions/AgentExtensions.cs (hunk 1 above)
Comment (Badulundabad) [reply CSharpSourceCode/Utilities/Extensions/AgentExtensions.cs:388]: position or direction or both?

Comment (Badulundabad) [reply CSharpSourceCode/Utilities/Extensions/AgentExtensions.cs:388]: normalized direction

File: CSharpSourceCode/Battle/Damage/TestingDamageMissionLogic.cs [hunk 2]
```diff
@@ -8,30 +8,46 @@ namespace TOW_Core.Battle.Damage
...
+            if (Mission.Scene.SlowMotionMode && _slowMotionEndTime <= Mission.CurrentTime)
             {
-                TOWDebug.EquipWeapon(Agent.Main, "tor_empire_weapon_gun_blunderbuss_001", EquipmentIndex.Weapon0);
-                TOWDebug.EquipWeapon(Agent.Main, "tor_empire_weapon_grenade_grenade", EquipmentIndex.Weapon1);
-                TOWDebug.EquipWeapon(Agent.Main, "tor_empire_weapon_grenade_shrapnel", EquipmentIndex.Weapon2);
+                Mission.Scene.SlowMotionMode = false;
             }
+        }
+
+        private void SomeZ3rcasMethod()
```
Comment (Z3rca) [inline CSharpSourceCode/Battle/Damage/TestingDamageMissionLogic.cs]: :D

**Review by Z3rca — APPROVED** [review]

File: CSharpSourceCode/Battle/Damage/TestingDamageMissionLogic.cs (hunk 2 above)
Comment (Badulundabad) [reply CSharpSourceCode/Battle/Damage/TestingDamageMissionLogic.cs]: fixed))

**Pull request merged**